server/tests/integration/test_dataset_upload.py

```diff
@@ -0,0 +1,64 @@
+import json
+from pathlib import Path
+import time
+
+from girder_client import GirderClient
+import pytest
+
+from .conftest import getClient, users
+
+
+def wait_for_jobs(client: GirderClient, max_wait_timeout=30):
+    """Wait for all worker jobs to complete"""
+    start_time = time.time()
+    incompleteJobs = []
+    while True and (time.time() - start_time < max_wait_timeout):
+        incompleteJobs = client.get(
+            'job',
+            parameters={
+                'statuses': json.dumps([0, 1, 2]),
+            },
+        )
+        if len(incompleteJobs) == 0:
+            break
+        time.sleep(1)
+    if len(incompleteJobs) > 0:
+        raise Exception("Jobs were still running after timeout")
+    # Verify that all jobs succeeded
+    did_not_succeed = client.get('job', parameters={'statuses': json.dumps([4, 5])})
+    if len(did_not_succeed) > 0:
+        raise Exception("Some jobs did not succeed")
+
+
+@pytest.mark.integration
+@pytest.mark.parametrize("user", users.values())
+@pytest.mark.run(order=2)
+def test_upload_user_data(user: dict):
```

Comment:
Wondering if there should be an intermediate step between `test_user_creation` and `test_upload_user_data` which will delete the public/private folders for the user?  That way tests can be run back to back to determine issues without destroying the volume or manually deleting the datasets in the folder.  If the folder already exists the test will fail.

Reply:
Great idea.